base/battle/Battle.ts

```diff
@@ -1102,6 +1102,80 @@ So, if a character will die after 5 turns and you land another Curse on them, it
                 }
             }
         }
+        let recovery = "";
+        //Check if allies can recover HP or PP at the turn's end and show it in the log
+        for (let i = 0; i < this.allies_info.length; i++) {
+            let player = this.allies_info[i];
+            if (player.instance.hp_recovery != 0) {
+                player.instance.current_hp = _.clamp(
+                    player.instance.current_hp + player.instance.hp_recovery,
+                    0,
+                    player.instance.max_hp
+                );
+                if (player.instance.hp_recovery > 0) {
+                    recovery = "recovers";
+                } else {
+                    recovery = "loses";
+                }
+                await this.battle_log.add(
+                    `${player.instance.name} ${recovery} ${Math.abs(player.instance.hp_recovery)} HP`
```

Comment:
The original game has a `!` at the end of the phrase.
![image](https://user-images.githubusercontent.com/7890647/103486219-014e0900-4ddb-11eb-80b7-28a048b37cd4.png)